### PR TITLE
Adds Loot Boxes

### PR DESCRIPTION
**Pull request by Ivniinvi**
## About The Pull Request
Adds BeeCoin lootboxes that can be purchased for 5000 BeeCoins. These will give you a random non-donor item from the shop. If you already own the item, you get half back.

## Why It's Good For The Game
The average BeeCoin cost of an item in the shop is 4494.92 BeeCoins, so this is on average a bad deal. The sense of pride and accomplishment definitely makes up for it though!

## Changelog
:cl:
add: Loot Boxes can now be purchased in the BeeCoin Shop under the OOC tab.
/:cl:

### CODE REVIEW

**Comment by DrMacCool** [comment]
If this is a joke, it's pretty funny. If not, I'm not really sure what this achieves. The large majority of BeeCoin Shop items are cheaper than 5000 coins, so it just seems like a way to gamble for the swag clothes, which kinda dilutes the effort that goes into unlocking it.

**Comment by Ivniinvi** [comment]
It's only half a joke. Lots of people will buy it because it tells people what they bought, even if statistically they're going to come out on the losing side, and everyone loves a good gamble.

**Comment by DrMacCool** [comment]
But what does being able to gamble OOC add to anything? The only reason loot boxes ever became a thing was because it was the easiest way for scum-bag video game designers use to squeeze as much money as possible out of the vulnerable. If you're not even scraping off the top, why would you want this? If you want gambling in SS13, play blackjack with people in-game or use the slot machines. You can get your fix in a way that only SS13 can, rather then porting in the most morally-bankrupt idea to ever grace the art form?

**Comment by Ivniinvi** [comment]
I want this because it adds fun for some people, and there is exactly zero money involved so there is no scum-baggery at all. Also, it's the worst way to get a specific item that you'd want, as you only have a 1 in 193 chance of getting any given item. A lot of people will use this either because they want to have fun, want to flex how many beecoins they have, or just legitimately don't know what to buy.

**Comment by Ivniinvi** [comment]
The loot box will now no longer send global messages while the round is ongoing.

**Comment by DrMacCool** [comment]
Exactly. If the odds are crap and it's nearly objective waste of money, why would you ever use the lootboxes and not just go back to grinding out the blessed antag token or the extra character slot? All I see is people losing out because they still fall into the same emotionally-taxing addictive loop that all loot boxes provide, with or without real-world dollars.

**Comment by DrMacCool** [comment]
It will be a funny meme for a week and then only the people who actually get hooked on it will ever use it again. Save it for next April if you really want to try it out.

**Comment by Ivniinvi** [comment]
Given the slow gain rate of BeeCoins (it would take 71 rounds of you completing your crew objective and surviving, or 25 greentext rounds), I don't really see people getting addicted to this? All I see is a fun way to spend your BeeCoins, that is 100% optional and only seeks to enhance fun.

**Comment by DrMacCool** [comment]
Perhaps I'm jaded because I've seen what this kind of thing does to people, but adding this in unironically is in bad taste, at least to me. Let's just see what the Discord has to say when it inevitably gets slapped in the feedback channel.

**Comment by checkraisefold** [comment]
add some valorant references and it's perfect

File: code/modules/client/loadout/loadout_ooc.dm [hunk 1]
```diff
@@ -15,8 +15,40 @@
...
 /datum/gear/ooc/real_antagtoken/purchase(var/client/C)
 	C.inc_antag_token_count(1)
 	message_admins("[C.ckey] has purchased a genuine antag token.")
 	log_game("[C.ckey] has purchased a genuine antag token.")
+
+/datum/gear/ooc/lootbox
+	display_name = "Loadout Loot Box"
+	description = "Gives a random item from the shop. If it's a duplicate, you receive half back. Free sense of pride and accomplishment included in every box."
+	cost = 5000
+	multi_purchase = TRUE
```
Comment (Archanial) [inline code/modules/client/loadout/loadout_ooc.dm:29]: You should make it 10k at least

File: code/modules/client/loadout/loadout_ooc.dm [hunk 2]
```diff
@@ -15,8 +15,40 @@
...
+
+/datum/gear/ooc/lootbox/purchase(var/client/C)
+	var/finding_item = TRUE
+	var/datum/gear/TG
```
Comment (Archanial) [inline code/modules/client/loadout/loadout_ooc.dm:33]: Tg? 😩

File: code/modules/client/loadout/loadout_ooc.dm [hunk 3]
```diff
@@ -15,8 +15,40 @@
...
+	while(finding_item)
```
Comment (Archanial) [inline code/modules/client/loadout/loadout_ooc.dm:34]: While without sleep

File: code/modules/client/loadout/loadout_ooc.dm [hunk 4]
```diff
@@ -15,8 +15,40 @@
...
+		TG = GLOB.gear_datums[pick(GLOB.gear_datums)]
+		if(!istype(TG, /datum/gear/donator) && !istype(TG, /datum/gear/ooc/lootbox))
```
Comment (Archanial) [inline code/modules/client/loadout/loadout_ooc.dm:36]: Use early returns

File: code/modules/client/loadout/loadout_ooc.dm [hunk 5]
```diff
@@ -15,8 +15,40 @@
...
+			finding_item = FALSE
+			if(TG.id in C.prefs.purchased_gear)
+				if(SSticker.current_state != 3)
+					to_chat(world, "<span class=boldannounce>[C.ckey] bought a lootbox, and got a [TG.display_name], but it was a duplicate.")
```
Comment (Archanial) [inline code/modules/client/loadout/loadout_ooc.dm:40]: This is bad idea

**Review by Archanial — CHANGES_REQUESTED** [review]
It's ultimately up to crossed

**Comment by Crossedfall** [comment]
This is better as an April Fools joke over an actual long-term change. 

Breaking down the two main issues I have:

1. OOC messages for people opening lootboxes is funny *maybe* once, but has the potential to be really annoying and could glorify the gambling practice.
2. These don't serve any actual function other than being a meme with openly terrible odds. Yes, they can act as a metacurrency sink, which we honestly kind of need, but, it doesn't actually add anything. Rather, it detracts from the overall shop by being purposefully terrible.

**Pull request closed without merging**